docs/Built-ins.md

```diff
@@ -13,6 +13,181 @@ These rules are available without any npm installation, via the `WORKSPACE` inst
 [labels]: https://bazel.build/docs/build-ref.html#labels
 
 
+## node_repositories
+
+To be run in user's WORKSPACE to install rules_nodejs dependencies.
+
+This rule sets up node, npm, and yarn.
+
+The versions of these tools can be specified in one of three ways:
+- Normal Usage:
+Specify no explicit versions. This will download and use the latest NodeJS & Yarn that were available when the
+version of rules_nodejs you're using was released.
+- Forced version(s):
+You can select the version of NodeJS and/or Yarn to download & use by specifying it when you call node_repositories,
+but you must use a value that matches a known version.
+- Using a custom version:
+You can pass in a custom list of NodeJS and/or Yarn repositories and URLs for node_resositories to use.
+- Using a local version:
+To avoid downloads, you can check in vendored copies of NodeJS and/or Yarn and set vendored_node and or vendored_yarn
+to point to those before calling node_repositories.
+
+This rule exposes the `@nodejs` workspace containing some rules the user can call later:
+
+- Run node: `bazel run @nodejs//:node path/to/program.js`
+- Install dependencies using npm: `bazel run @nodejs//:npm install`
+- Install dependencies using yarn: `bazel run @nodejs//:yarn`
+
+This rule also exposes the `@yarn` workspace for backwards compatibility:
+
+- Alternately install dependencies using yarn: `bazel run @yarn//:yarn`
+
+Note that the dependency installation scripts will run in each subpackage indicated by the `package_json` attribute.
+
+This approach uses npm/yarn as the package manager. You could instead have Bazel act as the package manager, running the install behind the scenes.
+See the `npm_install` and `yarn_install` rules, and the discussion in the README.
+
+Example:
+
+```
+load("@build_bazel_rules_nodejs//:defs.bzl", "node_repositories")
+node_repositories(package_json = ["//:package.json", "//subpkg:package.json"])
+```
+
+Running `bazel run @nodejs//:yarn` in this repo would create `/node_modules` and `/subpkg/node_modules`.
+
+
+
+### Usage
+
+```
+node_repositories(name, node_repositories, node_urls, node_version, package_json, preserve_symlinks, vendored_node, vendored_yarn, yarn_repositories, yarn_urls, yarn_version)
+```
+
+
+
+#### `name`
```

Comment:
This is used for toolchains; I don't think a user would ever want to specify this without blowing things up as the `host` is determined from the `name`:
```
    if repository_ctx.name == "nodejs":
        host = os_name(repository_ctx)
    else:
        host = repository_ctx.name.split("nodejs_", 1)[1]
```

Reply:
hmm yeah this is the one problem of doc'ing the rule instead of the macro. the macro sets the name... what should we do?